linchpin/examples/example5/README.md

```diff
@@ -0,0 +1,178 @@
+* `kstest_ftp_url` - FTP url of installation repository (required for `ftp` tests).
+* `kstest_result_run_dir_prefix` - prefix to be added to subdirectories storing *test run* results
+* `kstest_remote_results_path` - location of *results host*
+* `kstest_remote_results_keep_local` - do not keep results of nightly runs on *master*
+
+[schedule.yml](schedule.yml) contains modifications to the [default configuration](../../../ansible/roles/kstest-master/defaults/main/schedule.yml)
+
+
+#### The lifecycle:
+
+1) Provision the permanent *target*
+
+We are using existing [ssh key](../../README.md#ssh-keys) (`--key-use-existing`) that will need to be added to authorized keys of *restults host* in the next step. The path to the private key (`--ansible-private-key`) that will be used also as *master* key (`--key-use-for-master`) needs to be supplied so it can be added to the *master* during its deployment.
+
+```
+./kstests-in-cloud.sh provision nightly-permanent --pinfile examples/example5/PinFile --key-name kstests --key-use-existing --ansible-private-key ~/.ssh/kstests.pem --key-use-for-master
+```
+
+2) Add ssh key to *results host*
+
+The `kstests` public key needs to be added to *result host* so that *master* can rsync the results.
+
+3) Create test and scheduling configuration
+
+Copy the example configuration files to the ansible role's [location](../../../ansible/roles/kstest-master/vars/main) from which the role's defaults will be overriden.
+
+
+```
+cp linchpin/examples/example5/test-configuration.yml ansible/roles/kstest-master/vars/main/test-configuration.yml
+cp linchpin/examples/example5/schedule.yml ansible/roles/kstest-master/vars/main/schedule.yml
+```
+
+You may want to track the files in git repository (also to be aware that the default configuration will be actually overriden when running tests from this repository):
+```
+git add ansible/roles/kstest-master/vars/main/test-configuration.yml
+git add ansible/roles/kstest-master/vars/main/schedule.yml
+```
+
+4) Use the inventory generated with provisioning
+
+
+When applying the configuration for the first time it can be useful to set up the inventory for the local `kicstart-tests` repository so that it does not have to be passed to `ansible-playbook` with `-i` option each time the configuration is applied (or other playbook is run).
+
+To get the path of the `nightly-parmanent` *target*'s inventory run:
+
+```
+./kstests-in-cloud.sh status nightly-permanent
+```
+showing:
+```
+Target "nightly-permanent" has generated inventory linchpin/inventories/nightly-permanent.inventory
+```
+To configure ansible locally to use the inventory create local ansible configuration for the current directory:
+
+```
+echo -e "[defaults]\ninventory=linchpin/inventories/nightly-permanent.inventory" >  ansible.cfg
+```
+and perhaps add it to git for tracking
+```
+git add ansible.cfg
```

Comment:
Why? You can't push that to the upstream. I think this can be removed.

Reply:
Still waiting on answer.

Reply:
This is just for local tracking of configuration of a permanent target, using git when it is available for free.